src/Agent.Worker/Build/GitSourceProvider.cs

```diff
@@ -25,18 +25,26 @@ public class GitSourceProvider : SourceProvider, ISourceProvider
 
         public override string RepositoryType => WellKnownRepositoryTypes.Git;
 
-        public virtual async Task GetSourceAsync(IExecutionContext executionContext, ServiceEndpoint endpoint, CancellationToken cancellationToken)
+        public virtual async Task GetSourceAsync(
+            IExecutionContext executionContext,
+            ServiceEndpoint endpoint,
+            CancellationToken cancellationToken)
         {
             Trace.Entering();
+            // Validate args.
+            ArgUtil.NotNull(executionContext, nameof(executionContext));
             ArgUtil.NotNull(endpoint, nameof(endpoint));
 
             executionContext.Output($"Syncing repository: {endpoint.Name} (Git)");
             _gitCommandManager = HostContext.GetService<IGitCommandManager>();
             await _gitCommandManager.LoadGitExecutionInfo(executionContext);
 
-            string targetPath = executionContext.Variables.Get(Constants.Variables.Build.SourcesDirectory);
-            string sourceBranch = executionContext.Variables.Get(Constants.Variables.Build.SourceBranch);
-            string sourceVersion = executionContext.Variables.Get(Constants.Variables.Build.SourceVersion);
+            string targetPath;
+            string sourceBranch;
+            string sourceVersion;
+            endpoint.Data.TryGetValue(Constants.Variables.Build.SourcesDirectory, out targetPath);
```

Comment:
I think it would make more organization sense to create a separate set of constants specific to endpoint data rather than use the definition-variable-constants.

E.g.
```csharp
public static class Constants
{
    [...]
    public static class EndpointData
    {
        [...]
        public static readonly string SourceVersion = "SourceVersion";
    }
}